pyserum/market/core.py

```diff
@@ -0,0 +1,474 @@
+"""Market module to interact with Serum DEX."""
+from __future__ import annotations
+
+import itertools
+import logging
+from typing import List, Union
+
+from solana.account import Account
+from solana.publickey import PublicKey
+from solana.rpc.types import RPCResponse
+from solana.system_program import CreateAccountParams, create_account
+from solana.transaction import Transaction, TransactionInstruction
+from spl.token.constants import ACCOUNT_LEN, TOKEN_PROGRAM_ID, WRAPPED_SOL_MINT
+from spl.token.instructions import CloseAccountParams
+from spl.token.instructions import InitializeAccountParams, close_account, initialize_account
+
+from pyserum import instructions
+import pyserum.market.types as t
+
+from ..enums import OrderType, SelfTradeBehavior, Side
+from ..open_orders_account import OpenOrdersAccount, make_create_account_instruction
+from ..async_open_orders_account import AsyncOpenOrdersAccount
+from ._internal.queue import decode_event_queue
+from .orderbook import OrderBook
+from .state import MarketState
+
+LAMPORTS_PER_SOL = 1000000000
+
+
+# pylint: disable=too-many-public-methods
+class MarketCore:
+    """Represents a Serum Market."""
+
+    logger = logging.getLogger("pyserum.market.Market")
+
+    def __init__(self, market_state: MarketState, force_use_request_queue: bool = False) -> None:
+        self.state = market_state
+        self.force_use_request_queue = force_use_request_queue
+
+    def _use_request_queue(self) -> bool:
+        return (
+            # DEX Version 1
+            self.state.program_id == PublicKey("4ckmDgGdxQoPDLUkDT3vHgSAkzA3QRdNq5ywwY4sUSJn")
+            or
+            # DEX Version 1
+            self.state.program_id == PublicKey("BJ3jrUzddfuSrZHXSCxMUUQsjKEyLmuuyZebkcaFp2fg")
+            or
+            # DEX Version 2
```

Comment:
version 3?

Reply:
@leonardgee there are two public keys commented as v2 and none as v3, is this intentional?